### DISCUSSION TITLE
"Feed" is insufferable!

### DISCUSSION THREAD

**Discussion by jclxx** [community/community#62488] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

The thing I want on my home page is my repositories:

* First, the ones I created
* Second, the ones I've contributed to or are a member of in some way

I don't want anything else.

Honestly, I can't even find my repositories right now.  Doubtless I "just" click on some icon once I find it.

**Comment by ghost** [comment]
What the hell is this feed nonsense?

**Reply by ghost** [reply]
I actually have a deeper response to this now, which hopefully can be perceived as constructive criticism.

The feed *violates* my most efficient work mode.

The feed assumes a mode of working which is predominantly *reactive*. It assumes that having events related to tasks (not necessarily issues, but more generally things to do) listed as an entry point to work is the most useful (A), and furthermore assumes that having new events prioritized over existing ones is useful (B).

As to point A, this is not without merit - however, GitHub offers alternatives in the form of e.g. Kanban boards to tailor this to your own personal or project needs. A well curated board not only gives you a TODO list, it also provides feedback on progress, etc. If GH's default view was a Kanban board, I could understand the usefulness.

Instead, the selection of events is outside of the user's control. That means unlike Kanban board, there is no focus. There is no distinction between things that actually need doing, and things that just look like they might need the user's attention. There is no user-led curation.

Regarding B, having new events prioritized over existing ones is useful *only in highly specific contexts*, such as IT incident management. If some failure occurs, it's often important to triage it quickly. But development is very rarely related to such incident management. If anything, companies employ incident management processes for issue triage, after which they either become regular tasks for developers, or things to be dealt with immediately by an incident response team.

Only in resource constrained or badly managed companies do the responsibilities of incident response and regular development cross over regularly, as one work mode is not known to be compatible with the other.

Finally, regarding my own work: I tend to cross over between repositories a fair bit, but not cross over between all of them. As such I often create offline TODO lists for the day. Then it is fairly important for me to most easily reach a repository overview, for accessing information on the subset of repos that are relevant that day. The feed as the home screen simply complicates this by requiring more clicks on smaller surface areas, and providing zero use at the same time.

**Reply by jclxx** [reply]
Your views are interesting about reactive vs instigative tasks.  And when we're trying to instigate, the most precious thing we have is concentration and focus.  We're not idling about, looking for interesting things to distract us!

It's honestly amazing, isn't it, what Github has done here?  Does it really not understand its enormous userbase to this degree?  How did this get any kind of support?  How did it get past initial testing?

**Reply by ghost** [reply]
I suspect when the majority of your team is occupied with making small changes to an existing codebase, it can work.

**Comment by deligent4** [comment]
I dont know whose idea it was to put feed into something like github, there are plenty of sites available for socializing. Github should be spared from all these non sense. If anyone from Github is reading this, please note that this feed thing sucks. I would like my old home screen view back.

**Comment by jclxx** [comment]
When I teach others, I often get them to do spatial analysis, by counting pixels or fraction of the screen assigned to different things.  And colour-coding to see what's doing what function.  We ask them _what is it for?_ and _what does the user want to do?_

Good design is principally about knowing your audience.  What does this say about what Github thinks of us?

Here's the "Feed", taken this morning.  Bright green is something I've worked on in the last month; lighter greens are things to do with me but not recent.  bright red is completely uninteresting or completely outside my work.  Dimmer red might possibly be interesting.

**This screen has _nothing_ about the three repositories I commited to yesterday.  Not a single pixel.  Every single non-navigation pixel is either an old project or a complete distraction.  They are not even in "recent activity" (left panel).**

![image](https://github.com/community/community/assets/4455475/a20c75a2-0678-4473-b591-5384b8aee150)

**My repositories weren't available in the verb-like "menu" icon at top left; they are behind the noun-like "me" icon top right.**

I'm speechless about how terribly this design suits my needs.

**Comment by mway** [comment]
Separate from the question of "Should this exist?", I have no idea how feed items are selected and ranked. My feed is almost exclusively:

- Something from an org or user I've never seen
- Technology completely unrelated to anything I interact with on GitHub
- Languages that I do not view or contribute to (anywhere, let alone on GitHub)
- Releases for things that I am not following
- Releases for things that I viewed literally once
- Essentially zero things related to what I do, view, or contribute to regularly

In essence, it feels like enshittification.

**Reply by jclxx** [reply]
Exactly what I found a coloured red.  Somehow this feels like a programming team which made something just to justify its existence, not because of any need anywhere.  Also: to see how bad this is, you have to actually *be* the user seeing that these are completely useless.  As well as programming languages and operating systems I never use, there are programming languages and operating systems in the feed I've never even heard of.
